controller_manager/src/resource_manager.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <utility>
+
+#include "hardware_interface/components/component_info.hpp"
+#include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/actuator_hardware.hpp"
+#include "hardware_interface/sensor_hardware.hpp"
+#include "hardware_interface/system_hardware.hpp"
+
+#include "pluginlib/class_loader.hpp"
+
+#include "./resource_manager.hpp"
+
+namespace controller_manager
+{
+
+class ResourceStorage
+{
+  static constexpr const char * pkg_name = "hardware_interface";
+  static constexpr const char * actuator_interface_name =
+    "hardware_interface::ActuatorHardwareInterface";
+  static constexpr const char * sensor_interface_name =
+    "hardware_interface::SensorHardwareInterface";
+  static constexpr const char * system_interface_name =
+    "hardware_interface::SystemHardwareInterface";
+
+public:
+  ResourceStorage()
```

Comment:
This is nice concept :+1: 

It could work nicely to separate functionality and storage in the ResourceManger.